### PR TITLE
Acme nsone provider

### PR DESCRIPTION
**Pull request by alwaysjolley**
Adding DNS ACME support for NS1 vendor.

Having trouble on certificate creation:
`[CRITICAL] WORKER TIMEOUT`

certificate is added to pending certificates list correctly and next pass of `bin/lemur pending_certs fetch_all_acme` correctly generates the certificate on staging lets encrypt.

What am I missing?

### CODE REVIEW

File: lemur/plugins/lemur_acme/tests/test_acme.py [hunk 1]
```diff
@@ -263,12 +263,14 @@ def test_create_authority(self, mock_current_app):
     @patch("lemur.plugins.lemur_acme.plugin.current_app")
     @patch("lemur.plugins.lemur_acme.dyn.current_app")
     @patch("lemur.plugins.lemur_acme.cloudflare.current_app")
+    @patch("lemur.plugins.lemur_acme.nsone.current_app"
```
Comment (hosseinsh) [inline lemur/plugins/lemur_acme/tests/test_acme.py:266]: missgin ")"

**Comment by alwaysjolley** [comment]
Any updates on this? functionality works but I would like more information if I am missing something. Thanks, -jolley

**Comment by csine-nflx** [comment]
@alwaysjolley 

Thank you for sending this over. A couple thoughts:

Are you still having the WORKER TIMEOUT issue?  Does the timeout happen when a user creates the certificate through the web interface but it subsequently succeeds via celery job?

Would you mind writing unit tests for this?  See the test_powerdns.py for examples.

File: lemur/plugins/lemur_acme/nsone.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+        current_app.logger.debug(log_data)
+    except Exception as err:
+        sentry.captureException()
+        log_data["message"] = "Failed to Retrieve Zone Data"
+        log_data["error"] = err
+        current_app.logger.debug(log_data)
+        raise
+    for record in records:
+        zone = record['zone']
+        if record['primary']['enabled']:
```
Comment (mortenson) [inline lemur/plugins/lemur_acme/nsone.py:43]: For my use case, I need attestation for zones that are not enabled. Removing this line worked for me.

**Comment by hosseinsh** [comment]
hi @alwaysjolley, you are planing to work on this PR?

**Pull request closed without merging**